File: cpp/2283. 判断一个数的数字计数是否等于数位的值.cpp
class Solution {
public:
    bool digitCount(string num) {
        int n=num.size();
        int a[10]={0};
        for(int i=0;i<n;i++){
            a[num[i]-'0']++;
        }
        for(int i=0;i<n;i++){
            if(a[i]==(num[i]-'0')) continue;
            else {
                return false;
            }
        }
        return true;
    }
};

File: golang/2283. 判断一个数的数字计数是否等于数位的值.go
func digitCount(num string) bool {
    mp:=make(map[rune]int)
    for _,x:=range num{
        mp[x-'0']++
    }
    for i,x:=range num{
        if mp[rune(i)]==int(x-'0'){
            continue
        }else{
            return false
        }
    }
    return true
}